assignment2/assignment2.py

```diff
@@ -0,0 +1,418 @@
+
+    Note that there is no return statement in the lambda.  There is the parameter passed to the lambda (a row), followed by a colon, followed by the expression that gives the result.
+
+        Run the test to make sure things still work.
+
+    """
+
+    matches = list(
+        filter(
+            lambda row: int(row[employee_id_column]) == employee_id, employees["rows"]
+        )
+    )
+    return matches
+
+
+# Task 7
+def sort_by_last_name():
+    """We want to call the sort() method on the rows.  However, we need to tell it which column to use for the sort.
+
+    Create a function sort_by_last_name.  It takes no parameters.  You sort the rows you have stored in the dict.
+
+    Within the function, you call employees["rows"].sort().  This sorts the list of rows in place. But, you need pass to the list.sort() method a keyword argument called key (so you pass a parameter with key= when you call it).  You set that keyword parameter equal to a lambda.  The lambda is passed the row, and the expression after the colon gives the value from the row to be used in the sort.  You might want to use your column_index function for last_name so you know which value from the row should be given in the lambda expression.  Remember that the sort() method sorts the list in place and does not return the sorted list.
+
+    The sort_by_last_name function returns the sorted list of rows.
+
+    Run the test until this works.
+
+    Call the function in your program, and then print out the employees dict, to see it in sorted form.
+    """
+
+    employees["rows"].sort(key=lambda row: row[column_index("last_name")])
+
+    return employees["rows"]
+
+
+sort_by_last_name()
+print("\n\n", employees)
+
+
+# Task 8
+def employee_dict(row):
+    """
+
+        Create a function called employee_dict.  It is passed a row from the employees dict (not a row number).  It returns a dict.
+            The keys in the dict are the column headers from employees["fields"].
+            The values in the dict are the corresponding values from the row.
+            Do not include the employee_id in the dict. You skip that field for now.
+
+        Return the resulting dict for the employee.
+
+        Add a line to your program that calls this function and prints the result.  Use a row from the rows stored in the employees dict to pass to the function for this test.
+
+        Get the test working.
+
+    If you want to try something extra, look up the zip() function, which can be used to simplify the code for this problem.
+    """
+    employee = {}
+    for index, column in enumerate(employees["fields"]):
+        if column == "employee_id":
+            continue
+        employee[column] = row[index]
+
+    return employee
+
+    # Alternatively we can use the zip function
+    return {x: y for x, y in zip(employees["fields"], row) if x != "employee_id"}
+
```

Comment:
I like your attempt to use list comprehension here, but this return statement cannot be reached